### PR TITLE
Chuwong/enhance-ui

### PR DESCRIPTION
**Pull request by chuwong35122**
- Fix: layer generation bug when there is 1 asset in a layer
- Improve a GENEROUS amount of UI in ticket creation

### CODE REVIEW

File: components/generate/complete/CompleteAssetPreviewCard.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      <motion.div variants={variant}>
+        <Box
+          sx={{
+            position: "relative",
+            maxHeight: "340px",
+            width: "240px",
+            overflow: "hidden",
+            border: 2,
+            marginRight: 2,
+            marginBottom: 2,
```
Comment (pointmekin) [inline components/generate/complete/CompleteAssetPreviewCard.tsx:84]: this is ok, but this is cooler
```
mb: 2,
mr: 2
```

File: components/generate/complete/steps/CreateCompleteTicketStep2.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          {uploadedAssets.length > 0 && (
+            <Box
+              sx={{
+                paddingY: 2,
+                maxHeight: "660px",
+                overflowY: "auto"
+              }}
+            >
+              <Stack direction="row" alignItems="center" flexWrap="wrap">
+                {uploadedAssets.map((asset, i) => (
```
Comment (pointmekin) [inline components/generate/complete/steps/CreateCompleteTicketStep2.tsx:51]: would be nice to add type here

File: utils/generate/layer.ts [hunk 3]
```diff
@@ -53,6 +66,7 @@ export function generateRandomLayer(
     }
   }
 
+  // console.table(metadata)
```
Comment (pointmekin) [inline utils/generate/layer.ts:69]: lmao

File: utils/generate/layer.ts [hunk 4]
```diff
@@ -27,13 +27,26 @@ export function generateRandomLayer(
...
-      const rndIndex = weightedRandom(occurrences, layer.assets)
-      assetAttributes.push({
-        layerId: layer.layerId,
-        layerName: layer.layerName,
-        layerOccurrence: layer.layerOccurrence,
-        asset: layer.assets[rndIndex]
-      })
+      if (layer.assets.length > 1) {
+        // requires checking since weightedRandom does not account for assets length of 1
+        const rndIndex = weightedRandom(occurrences, layer.assets)
```
Comment (pointmekin) [inline utils/generate/layer.ts:32]: I notice you have cases where error can be thrown in `weightedRandom`.
Do you need to add this into a trycatch?

Comment (chuwong35122) [reply utils/generate/layer.ts:32]: will create alert later

**Pull request merged**